diesel/src/pg/expression/helper_types.rs

```diff
@@ -352,3 +352,8 @@ pub type range_merge<R1, R2> = super::functions::range_merge<SqlTypeOf<R1>, SqlT
 #[allow(non_camel_case_types)]
 #[cfg(feature = "postgres_backend")]
 pub type array_append<A, E> = super::functions::array_append<SqlTypeOf<A>, SqlTypeOf<E>, A, E>;
+
+/// Return type of [`array_ndims(array)`](super::functions::array_ndims())
+#[allow(non_camel_case_types)]
+#[cfg(feature = "postgres_backend")]
+pub type array_ndims<A> = super::functions::array_ndims<SqlTypeOf<A>, A>;
```

Comment:
Is this even required for functions that do not have a generic return type and always return `Integer`? 
I tried to remove it but without this generic arguments in `auto_types `explodes.

Reply:
It's required for the generic arguments. `define_sql_function!` generates a function signature as follows:

```
fn array_ndims<A, Arr>(arr: Arr) -> Integer
where Arr: AsExpression<A>, A: ArrayOrNullableArray + SingleValue>;
```

The `AsExpression` part allows us to automatically convert rust values (i.e. `vec![1,2,3]`) to sql side binds and at the same time it allows us to accept sql side expressions with the same type at the same location (i.e `table::array_column`).  On the other hand the `#[auto_type]` macro expects exactly as many generic arguments as there are function parameters. The type def in then used to map that.